### PR TITLE
Update core version to 4.5.0 and language to 1.13

### PR DESCRIPTION
**Pull request by vmatyusGitHub**
## PR Checklist

Please make sure that your PR fulfills the following requirements:

- [ ] The commit message follows the
[Angular Commit Message Guidelines](https://github.com/angular/angular/blob/master/CONTRIBUTING.md#-commit-message-guidelines).
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type  

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] New tests
- [ ] Build/CI related changes
- [ ] Documentation content changes
- [ ] Other (please describe)

## What is the current behavior?

Update go core version from 4.3.0 to 4.5.0
Update language version from 1.14 to 1.15

## What is the new behavior?

* Set the expected language version for the module to 1.15
* resolve path params by name, not by parameter ordering
* IAM Authenticator: add support for optional 'scope' property
* iam/cp4d token refresh logic
* expose AuthenticationError.Err field and add ctor

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Review by ricellis — APPROVED** [review]

File: go.mod [hunk 1]
```diff
@@ -1,9 +1,9 @@
 module github.com/IBM/cloudant-go-sdk
 
-go 1.14
+go 1.15
```
Comment (bessbd) [inline go.mod:3]: What is the exact meaning of this line?
https://golang.org/ref/mod#go-mod-file-go tells me it's the expected language version. https://stackoverflow.com/a/58205552/5323166 and https://www.reddit.com/r/golang/comments/b0b8k3/what_version_should_the_gomod_go_directive_be_set/?utm_source=share&utm_medium=web2x&context=3 say it's meant as the minimum accepted version which according to https://github.com/IBM/cloudant-go-sdk#prerequisites is 1.13 now.

**Comment by bessbd** [comment]
Also, I understand this change fixes an issue of handling `/` characters in path parameters. Is that correct?

File: go.mod (hunk 1 above)
Comment (vmatyusGitHub) [reply go.mod]: From: https://golang.org/ref/mod#go-mod-file-go 
> The language version determines which language features are available when compiling packages in the module.

I thought it is better to install only one version from go to the desktop. This minimum accepted version aspect is new for me. I am okay to set it to 1.13. Change in 4b17402

**Review by bessbd — APPROVED** [review]

**Comment by vmatyusGitHub** [comment]
> Also, I understand this change fixes an issue of handling `/ `characters in path parameters. Is that correct?

Yes, this core version update and also generator update also needed for the correction.

**Comment by vmatyusGitHub** [comment]
Thank you for the review @ricellis and @bessbd ! I am about to merge this PR!

**Pull request merged**